applications/ConstitutiveLawsApplication/custom_processes/set_automated_initial_damage_process.cpp

```diff
@@ -0,0 +1,133 @@
+
+        KRATOS_TRY
+
+        const array_1d<double, 3> hole_generatrix_axis = mThisParameters["hole_generatrix_axis"].GetVector();
+        KRATOS_ERROR_IF(MathUtils<double>::Norm3(hole_generatrix_axis) < machine_tolerance) << "The hole generatrix axis has norm zero" << std::endl;
+        
+        const array_1d<double, 3> hole_generatrix_point = mThisParameters["hole_generatrix_point"].GetVector();
+        auto& process_info = mrThisModelPart.GetProcessInfo();  
+
+        array_1d<double, 3> normalized_generatrix_vector;
+        normalized_generatrix_vector[0] = (hole_generatrix_axis[0] * hole_generatrix_axis[0]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+        normalized_generatrix_vector[1] = (hole_generatrix_axis[1] * hole_generatrix_axis[1]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+        normalized_generatrix_vector[2] = (hole_generatrix_axis[2] * hole_generatrix_axis[2]) / std::sqrt(hole_generatrix_axis[0] * hole_generatrix_axis[0] + hole_generatrix_axis[1] * hole_generatrix_axis[1] + hole_generatrix_axis[2] * hole_generatrix_axis[2]);
+
+        const double hole_radius_offset = mThisParameters["hole_radius_offset"].GetDouble();
+
+        int table_id = mThisParameters["table_id"].GetInt();
+
+        block_for_each(mrThisModelPart.Elements(), [&](auto& rElement) {
+
+            const array_1d<double, 3>& r_element_centroid = rElement.GetGeometry().Center();
+            
+            array_1d<double, 3> relative_position_vector;
+            relative_position_vector = r_element_centroid - hole_generatrix_point;
+
+            double vector_scaler = MathUtils<double>::Dot3(relative_position_vector, normalized_generatrix_vector);
+
+            const array_1d<double, 3> intersection_point = hole_generatrix_point + vector_scaler * normalized_generatrix_vector;
+
+            const array_1d<double, 3> radial_position_vector = r_element_centroid - intersection_point;
+
+            double centroid_relative_distance = std::sqrt(radial_position_vector[0] * radial_position_vector[0] + radial_position_vector[1] * radial_position_vector[1] + radial_position_vector[2] * radial_position_vector[2]) - hole_radius_offset;
+
+
+            if (centroid_relative_distance < 0.0){
+                if (std::abs(centroid_relative_distance) <= tolerance) {
+                    centroid_relative_distance = 0.0;
+                }
+                else {
+                    KRATOS_ERROR << "The relative centroid distance may not be negative. Check the hole radius offset and the thickness of element " << rElement.Id() << std::endl;
+                }
+            }
+
+            double initial_damage_value = mrThisModelPart.GetTable(table_id).GetValue(centroid_relative_distance);
```

Comment:
If not changing const (idem for the rest)

Reply:
It may change in lines 86 and 89.